### PR TITLE
58 allow users to create events and edit their existing events in app

### PR DESCRIPTION
**Pull request by Sundance636**
Please review and give plenty of feedback,

Some things to note:
- my initial plan with the having separate fragments for both adding and editing fragment, didn't turn out exactly as i wanted cause of some thing that i couldn't so with the xml files, so those two fragments have a lot of similar code, I it at least "separates concerns"(they still work perfectly fine though), but if you guys think we should just have one AddEditFragment let me know.
- if you're going to be testing, probably easier to just start the app from the events list page(to avoid making new user and being unable to edit/see your own events), and make sure you load your user too first.

### CODE REVIEW

**Comment by Logan9312** [comment]
I cannot save events to the database, when I click Save from the add event screen, or Edit from the event screen I get the following error:

2024-03-06 09:43:53.234  5054-5263  StorageUtil             com.example.scanpal                  W  Error getting App Check token; using placeholder token instead. Error: com.google.firebase.FirebaseException: No AppCheckProvider installed.
2024-03-06 09:43:53.235  5054-5263  NetworkRequest          com.example.scanpal                  W  no auth token for request

Is this happening to anyone else?

File: app/src/main/java/com/example/scanpal/AddEventFragment.java [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.navigation.NavController;
+import androidx.navigation.fragment.NavHostFragment;
+
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
+import com.google.firebase.firestore.FirebaseFirestore;
+
+public class AddEventFragment extends Fragment {
```
Comment (Logan9312) [inline app/src/main/java/com/example/scanpal/AddEventFragment.java:28]: Docstring

File: app/src/main/java/com/example/scanpal/EditEventFragment.java [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.navigation.NavController;
+import androidx.navigation.fragment.NavHostFragment;
+
+import com.bumptech.glide.Glide;
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
+import com.google.firebase.firestore.FirebaseFirestore;
+
+public class EditEventFragment extends Fragment {
```
Comment (Logan9312) [inline app/src/main/java/com/example/scanpal/EditEventFragment.java:29]: Docstring

**Review by Logan9312 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/scanpal/EventDetailsFragment.java [hunk 3]
```diff
@@ -25,13 +29,20 @@
 public class EventDetailsFragment extends Fragment {
```
Comment (Logan9312) [inline app/src/main/java/com/example/scanpal/EventDetailsFragment.java:29]: Docstring

**Review by Logan9312 — CHANGES_REQUESTED** [review]

**Review by Nandan-18 — APPROVED** [review]
lgtm! should be good to merge once Logan's comments are addressed.

**Review by Logan9312 — APPROVED** [review]

**Pull request merged**